OrbitCore/CallstackData.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CORE_CALLSTACK_DATA_H_
+#define ORBIT_CORE_CALLSTACK_DATA_H_
+
+#include <memory>
+
+#include "BlockChain.h"
+#include "Callstack.h"
+#include "CallstackTypes.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/synchronization/mutex.h"
+#include "capture_data.pb.h"
+
+class CallstackData {
+ public:
+  explicit CallstackData() = default;
+
+  void AddCallstackEvent(orbit_client_protos::CallstackEvent&& callstack_event);
+  void AddUniqueCallStack(CallStack call_stack);
+  void AddCallStackFromCallstackData(
+      const orbit_client_protos::CallstackEvent& event,
+      const CallstackData& other);
+
+  [[nodiscard]] const BlockChain<orbit_client_protos::CallstackEvent,
+                                 16 * 1024>&
+  GetCallstackEvents() const { return callstack_events_; };
```

Comment:
For simple getters, the name should be `callstack_events()`.
For setters it would be `set_callstack_events`.
https://google.github.io/styleguide/cppguide.html#Function_Names

Reply:
Also, can you add  [[nodiscard]]

Reply:
I thought that it's not a strict rule (so you may use one of two: GetCallstackEvents() or callstack_events()). For which function do you mean add `[[nodiscard]]`?

Reply:
I though we wanted to make it more strict. I would add them to all getters here.
Basically, I would add them always, when you can consider the behavior of the program to be incorrect, when calling the method and dropping the return value.
In this case: Calling a getter and not using the value, can only be wrong. Thus, you want the compiler to tell you this.

Reply:
Renamed the function